context/app/static/js/pages/Genes/hooks.js

```diff
@@ -26,4 +26,55 @@ function useGeneCommonName(geneSymbol) {
   return { geneCommonName, isLoading };
 }
 
-export { useGeneCommonName };
+async function fetchGeneData(geneDataURL) {
+  const response = await fetch(geneDataURL);
+
+  if (!response.ok) {
+    console.error('Gene data API failed.', response);
+  }
+
+  const responseText = await response.text();
+  const result = await parseStringPromise(responseText);
+  const geneSummary = result['Entrezgene-Set'].Entrezgene[0].Entrezgene_summary[0];
+  return geneSummary;
+}
+
+const useGeneData = (geneID) => {
```

Comment:
- The only thing we currently need the `geneId` for is to provide it to the `useGeneData` hook
 - To start fetching `geneSummary` we need `geneId` to be available, but `useGeneData` currently doesn't do anything to prevent an attempted fetch when `geneId` is not yet available

My first suggestion is to modify the `useGeneData` hook to prevent it from trying to fetch data when it's being provided with a blank `geneId` ( since `geneID` is `''` until the first hook finishes fetching). The [Conditional Fetching](https://swr.vercel.app/docs/conditional-fetching) section of the SWR documentation should be helpful for this.

Additionally, we should consider using the `useGeneId` hook inside the `useGeneData` hook and making the `useGeneData` hook take in the `geneSymbol` instead. The `geneID` would still be internally cached by SWR for any other queries which may rely on it in the future, and the `useGeneData` hook would be a bit easier to reuse

Reply:
This all makes sense, thanks so much for the feedback!